src/events/IonEventStream.ts

```diff
@@ -93,27 +107,39 @@ export class IonEventStream {
                     writer.stepIn(tempEvent.ionType!);
                     break;
                 case IonEventType.CONTAINER_END:
+                    if(isEmbedded) {
+                        isEmbedded = false;
+                    }
                     writer.stepOut();
                     break;
                 case IonEventType.STREAM_END:
-                    writer.close();
                     break;
                 case IonEventType.SYMBOL_TABLE:
                     throw new Error("Symboltables unsupported.");
                 default:
                     throw new Error("Unexpected event type: " + tempEvent.eventType);
-
             }
         }
+        writer.close();
     }
 
     getEvents(): IonEvent[] {
         return this.eventStream;
     }
 
-    equals(expected: IonEventStream) {
+    equals(expected: IonEventStream): boolean {
+        return this.compare(expected).result == ComparisonResultType.EQUAL;
+    }
+
+    /**
+     *  compares eventstreams and generates comparison result
+     */
+    compare(expected: IonEventStream): ComparisonResult {
         let actualIndex: number = 0;
         let expectedIndex: number = 0;
+        if(this.eventStream.length != expected.eventStream.length) {
+            return new ComparisonResult(ComparisonResultType.ERROR, "Both event streams of different length");
```

Comment:
```suggestion
            return new ComparisonResult(ComparisonResultType.ERROR, "The event streams have different lengths");
```

Also: why is this an `ERROR` rather than just `NOT_EQUAL`?

Reply:
Yes, I think that is correct it should be NOT_EQUAL.